src/start.sh

```diff
@@ -53,11 +53,33 @@ start() {
   #migrate Gravity Database if needed:
   migrate_gravity
 
-  # Start pihole-FTL
-  start_ftl
+  echo "  [i] pihole-FTL pre-start checks"
+  # Remove possible leftovers from previous pihole-FTL processes
+  rm -f /dev/shm/FTL-* 2>/dev/null
+  rm -f /run/pihole/FTL.sock
 
-  # Give FTL a couple of seconds to start up
-  sleep 2
+  fix_capabilities
+  sh /opt/pihole/pihole-FTL-prestart.sh
+
+  echo "  [i] Starting pihole-FTL ($FTL_CMD) as ${DNSMASQ_USER}"
+  echo ""
+
+  capsh --user=$DNSMASQ_USER --keep=1 -- -c "/usr/bin/pihole-FTL $FTL_CMD >/dev/null" &
+  # Notes on above:
+  # - DNSMASQ_USER default of pihole is in Dockerfile & can be overwritten by runtime container env
+  # - /var/log/pihole/pihole*.log has FTL's output that no-daemon would normally print in FG too
+  #   prevent duplicating it in docker logs by sending to dev null
+  ftl_pid=$!
+
+  # Wait until the log file exists before continuing
+  while [ ! -f /var/log/pihole/FTL.log ]; do
+    sleep 0.5
+  done
+
+  #  Wait until the FTL log contains the "FTL started" message before continuing
+  while ! grep -q '########## FTL started' /var/log/pihole/FTL.log; do
+    sleep 0.5
+  done
```

Comment:
Looking again at:
```
pihole  | 2024-10-06 22:42:42.417 BST [56M] INFO: ########## FTL started on 0d9149f7f785! ##########
```

A working solution may be reading the PID from the PID file and expecting this in the log, like
```suggestion
  #  Wait until the FTL log contains the "FTL started" message before continuing
  FTL_PID_FILE="$(getFTLConfigValue files.pid)"
  FTL_PID="$(cat "${FTL_PID_FILE}")"

  while ! grep -q "[${FTL_PID}M] INFO: ########## FTL started" /var/log/pihole/FTL.log; do
    sleep 0.5
  done
```

Reply:
I've tried a few times to get this grep working (I had to put the `FTL_PID=` part into a while loop until it was populated otherwise `FTL_PID` was blank - however the grep didn't work, and I don't really know why!